UniWatch/DataAccess/LectureManager.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+using UniWatch.Models;
+
+namespace UniWatch.DataAccess
+{
+    /// <summary>
+    /// Provides central access to Lecture related information
+    /// </summary>
+    public class LectureManager : ILectureManager
+    {
+        private bool disposed = false;
+        private AppDbContext _db;
+
+        /// <summary>
+        /// Creates a default LectureManager
+        /// </summary>
+        public LectureManager()
+        {
+            _db = new AppDbContext();
+        }
+
+        /// <summary>
+        /// Creates a LectureManager with the given context
+        /// </summary>
+        /// <param name="context">The context to create the manager with</param>
+        public LectureManager(AppDbContext context)
+        {
+            _db = context;
+        }
+
+        /// <summary>
+        /// Returns the lecture with the given id
+        /// </summary>
+        /// <param name="lectureId">The id of the lecture</param>
+        /// <returns>The lecture with the given id</returns>
+        public Lecture Get(int lectureId)
+        {
+            return _db.Lectures.Find(lectureId);
+        }
+
+        /// <summary>
+        /// Returns a list of all lectures for the given class
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <returns>A list of all lectures for the class</returns>
+        public IEnumerable<Lecture> GetTeacherReport(int classId)
+        {
+            return _db.Lectures.
+                Where(lecture => lecture.Class.Id == classId)
+                .ToList();
+        }
+
+        /// <summary>
+        /// Returns a list of attendance for a student in a class
+        /// </summary>
+        /// <param name="classId">The id of the class</param>
+        /// <param name="studentId">The id of the student</param>
+        /// <returns>The list of attendance for the student in the class</returns>
+        public IEnumerable<StudentAttendance> GetStudentReport(int classId, int studentId)
+        {
+            return _db.Attendance
+                .Where(a => a.Lecture.Class.Id == classId && a.Student.Id == studentId)
+                .ToList();
+        }
+
+        /// <summary>
+        /// Updates and saves the given lecture
+        /// </summary>
+        /// <param name="lecture">The lecture to update</param>
+        /// <returns>The updated lecture</returns>
+        public Lecture Update(Lecture lecture)
+        {
+            var lect = _db.Lectures.Find(lecture.Id);
```

Comment:
`lect` is never used. Maybe you meant to use `lect` instead of `lecture` in some places.